tests/langchain/test_langchain_autolog.py

```diff
@@ -686,19 +687,106 @@ def test_combine_input_and_output(input, output, expected):
+    assert run_data.tags["mlflow.autologging"] == "langchain"
     artifacts = get_artifacts(run.info.run_id)
     for artifact_name in get_mlflow_callback_artifacts():
         if isinstance(artifact_name, str):
             assert artifact_name in artifacts
         else:
             assert any(artifact_name.match(artifact) for artifact in artifacts)
+    # make sure INFERENCE_FILE_NAME is logged
+    mlflow.load_table(INFERENCE_FILE_NAME, run_ids=[run.info.run_id])
+    if isinstance(callbacks, BaseCallbackManager):
+        assert callbacks.handlers[0].logs == ["chain_start", "chain_end"]
+        callbacks.handlers[0].reset()
+    else:
+        assert callbacks[0].logs == ["chain_start", "chain_end"]
+        callbacks[0].reset()
+
+
+@pytest.mark.parametrize("invoke_arg", ["args", "kwargs"])
+@pytest.mark.parametrize(
+    "config",
+    [
+        RunnableConfig(callbacks=[CustomCallbackHandler()]),
+        RunnableConfig(callbacks=BaseCallbackManager([CustomCallbackHandler()])),
+        [
+            RunnableConfig(callbacks=[CustomCallbackHandler()]),
+            RunnableConfig(callbacks=[CustomCallbackHandler()]),
+        ],
+        [
+            RunnableConfig(callbacks=BaseCallbackManager([CustomCallbackHandler()])),
+            RunnableConfig(callbacks=BaseCallbackManager([CustomCallbackHandler()])),
+        ],
+    ],
+)
+def test_langchain_autolog_callback_injection_in_batch(invoke_arg, config):
+    mlflow.langchain.autolog(
+        log_models=True, extra_tags={"test_tag": "test"}, log_inputs_outputs=True
+    )
+    with mlflow.start_run() as run, _mock_request(return_value=_mock_chat_completion_response()):
+        model = create_openai_llmchain()
+        inputs = ["MLflow"] * 2
+        if invoke_arg == "args":
+            model.batch(inputs, config)
+        elif invoke_arg == "kwargs":
+            model.batch(inputs, config=config)
+        assert mlflow.active_run() is not None
+    run_data = MlflowClient().get_run(run.info.run_id).data
+    assert run_data.metrics["chain_starts"] == 2
+    assert run_data.metrics["chain_ends"] == 2
+    assert run_data.tags["test_tag"] == "test"
+    assert run_data.tags["mlflow.autologging"] == "langchain"
+    artifacts = get_artifacts(run.info.run_id)
+    for artifact_name in [f"chain_start_{i}.json" for i in range(1, 3)] + [
+        f"chain_end_{i}.json" for i in range(1, 3)
+    ]:
+        assert artifact_name in artifacts
+    # make sure INFERENCE_FILE_NAME is logged
+    mlflow.load_table(INFERENCE_FILE_NAME, run_ids=[run.info.run_id])
+    if isinstance(config, list):
+        callbacks = config[0]["callbacks"]
+        expected_logs = sorted(["chain_start", "chain_end"])
+    else:
+        callbacks = config["callbacks"]
+        expected_logs = sorted(["chain_start", "chain_end"] * 2)
+    if isinstance(callbacks, BaseCallbackManager):
+        assert sorted(callbacks.handlers[0].logs) == expected_logs
+        callbacks.handlers[0].reset()
```

Comment:
What is this line for?

Reply:
To clean up the logs field, otherwise the same object can be reused across tests (for args/kwargs) and the result is not correct

Reply:
Got it. We can make `config` a function so each test creates a new `CustomCallbackHandler` instance.

Reply:
See https://github.com/mlflow/mlflow/pull/11798#discussion_r1577302818